src/main/controllers/AboutHearingDocumentsController.ts

```diff
@@ -0,0 +1,164 @@
+import { Response } from 'express';
+
+import { Form } from '../components/form/form';
+import { isFieldFilledIn } from '../components/form/validator';
+import { AppRequest } from '../definitions/appRequest';
+import { TranslationKeys } from '../definitions/constants';
+import { FormContent, FormFields } from '../definitions/form';
+import { AnyRecord } from '../definitions/util-types';
+import { getLogger } from '../logger';
+
+import { assignFormData, getPageContent } from './helpers/FormHelpers';
+import { getLanguageParam } from './helpers/RouterHelpers';
+
+const logger = getLogger('AboutHearingDocumentsController');
+
+export default class AboutHearingDocumentsController {
+  private form: Form;
+  private aboutHearingDocumentsContent: FormContent;
+
+  constructor() {
+    // intentionally empty
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    const { userCase } = req.session;
+    req.session.errors = [];
+    const errorsForPage = () => {
+      if (!req.body.hearingDocumentsAreFor) {
+        req.session.errors.push({ propertyName: 'hearingDocumentsAreFor', errorType: 'required' });
+      } else {
+        userCase.hearingDocumentsAreFor = req.body.hearingDocumentsAreFor;
+      }
+      if (!req.body.whoseHearingDocumentsAreYouUploading) {
+        req.session.errors.push({ propertyName: 'whoseHearingDocumentsAreYouUploading', errorType: 'required' });
+      } else {
+        userCase.whoseHearingDocumentsAreYouUploading = req.body.whoseHearingDocumentsAreYouUploading;
+      }
+      if (!req.body.whatAreTheseDocuments) {
+        req.session.errors.push({ propertyName: 'whatAreTheseDocuments', errorType: 'required' });
+      } else {
+        userCase.whatAreTheseDocuments = req.body.whatAreTheseDocuments;
+      }
+    };
+
+    errorsForPage();
+    if (req.session.errors.length) {
+      return res.redirect('/about-hearing-documents');
+    }
+    return res.redirect('/');
+  };
+
+  public get = async (req: AppRequest, res: Response): Promise<void> => {
+    const formatDate = (rawDate: Date): string =>
+      new Intl.DateTimeFormat('en-GB', {
+        day: 'numeric',
+        month: 'short',
+        year: 'numeric',
+      }).format(new Date(rawDate));
+
+    if (!req.session?.userCase?.hearingCollection?.length) {
+      logger.info('no hearing collection found, redirecting to citizen hub');
+      return res.redirect(`/citizen-hub/${req.session.userCase.id}${getLanguageParam(req.url)}`);
+    }
+    const radioBtns = req.session?.userCase?.hearingCollection
```

Comment:
Same with this, please move to a helper class

Reply:
this has been moved.